src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/ObjectExplorerUtils.cs

```diff
@@ -53,7 +51,7 @@ public static bool VisitChildAndParents(TreeNode child, Predicate<TreeNode> visi
         /// determines whether to stop going further up the tree</param>
         /// <param name="filter">Predicate function to filter the children when traversing</param>
         /// <returns>A Tree Node that matches the condition, or null if no matching node could be found</returns>
-        public static TreeNode? FindNode(TreeNode node, Predicate<TreeNode> condition, Predicate<TreeNode> filter, bool expandIfNeeded = false)
+        public static TreeNode? FindNode(TreeNode node, Predicate<TreeNode> condition, Predicate<TreeNode> filter, bool expandIfNeeded = false, CancellationToken cancellationToken = new CancellationToken())
```

Comment:
Would suggest adding new functionality in separate PRs in the future, hard to efficiently review what's actual changes vs what's just moving stuff around.